--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -943,7 +943,7 @@
 
     solves++;
 
-    max_learnts               = nClauses() * learntsize_factor;
+    
     learntsize_adjust_confl   = learntsize_adjust_start_confl;
     learntsize_adjust_cnt     = (int)learntsize_adjust_confl;
     lbool   status            = l_Undef;
@@ -967,6 +967,7 @@
 
     if (verbosity >= 1)/*auto*/{
         printf("c ===============================================================================\n");
+        int i, j;
 }/*auto*/
 
 
@@ -1069,6 +1070,8 @@
     if (verbosity > 0)/*auto*/{
         printf("Wrote %d clauses with %d variables.\n", cnt, max);
 }/*auto*/
+
+    Clause& c = ca[clauses[i]];
 }
 
 
--- before: minisat.params
+++ after: minisat.params
@@ -1,6 +1,6 @@
-lbd-cut := 5.0
+lbd-cut := 4.932627441237257
 lbd-cut-max := 10.0
-cp-increase := 15000
+cp-increase := 5724
 core-tolerance := 0.02
 var-decay := 0.95
 cla-decay := 0.999
@@ -10,12 +10,12 @@
 rnd-init := 'False'
 luby := 0
 gc-frac := 0.2
-verb := 1
+verb := 0
 pre := 'True'
 asymm := 'False'
 rcheck := 'False'
 elim := 'False'
-grow := 0
-cl-lim := 20
+grow := -149
+cl-lim := 212
 sub-lim := 1000
 simp-gc-frac := 0.5
