--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -418,6 +418,10 @@
             }
         }
     }else/*auto*/{
+        if (L < LBD_cut){
+                lF.push(confl);
+                core_added++;
+            }
         i = j = out_learnt.size();
 }/*auto*/
 
--- before: minisat.params
+++ after: minisat.params
@@ -1,16 +1,16 @@
 lbd-cut := 5
-lbd-cut-max := 10
-cp-increase := 15000
-core-tolerance := 0.02
-R-val := 1.4
+lbd-cut-max := 22
+cp-increase := 6155
+core-tolerance := 0.5603469984284788
+R-val := 1.0884271733737099
 var-decay := 0.8
-cla-decay := 0.999
+cla-decay := 0.27564752676211635
 rnd-freq := 0.0
 ccmin-mode := '2'
 phase-saving := '2'
 rnd-init := 'False'
 luby := 0
-gc-frac := 0.2
+gc-frac := 0.5290205395201333
 verb := '1'
 pre := 'True'
 asymm := 'False'
@@ -18,5 +18,5 @@
 elim := 'True'
 grow := 0
 cl-lim := 20
-sub-lim := 1000
+sub-lim := 0
 simp-gc-frac := 0.5
