--- before: minisat.params
+++ after: minisat.params
@@ -1,6 +1,6 @@
 lbd-cut := 5
 lbd-cut-max := 10
-cp-increase := 15000
+cp-increase := 5146
 core-tolerance := 0.02
 R-val := 1.4
 var-decay := 0.8
@@ -10,8 +10,8 @@
 phase-saving := '2'
 rnd-init := 'False'
 luby := 0
-gc-frac := 0.2
-verb := '1'
+gc-frac := 0.6655879881293355
+verb := '2'
 pre := 'True'
 asymm := 'False'
 rcheck := 'False'
@@ -19,4 +19,4 @@
 grow := 0
 cl-lim := 20
 sub-lim := 1000
-simp-gc-frac := 0.5
+simp-gc-frac := 0.8932004792372763
