--- before: minisat.params
+++ after: minisat.params
@@ -2,7 +2,7 @@
 lbd-cut-max := 10
 cp-increase := 15000
 core-tolerance := 0.02
-R-val := 1.4
+R-val := 1.9758704273045082
 var-decay := 0.8
 cla-decay := 0.999
 rnd-freq := 0.0
@@ -11,12 +11,12 @@
 rnd-init := 'False'
 luby := 0
 gc-frac := 0.2
-verb := '1'
+verb := '0'
 pre := 'True'
 asymm := 'False'
 rcheck := 'False'
 elim := 'True'
 grow := 0
-cl-lim := 20
+cl-lim := 28
 sub-lim := 1000
 simp-gc-frac := 0.5
