--- before: minisat.params
+++ after: minisat.params
@@ -1,6 +1,6 @@
-lbd-cut := 5.0
+lbd-cut := 3.793079767119533
 lbd-cut-max := 10.0
-cp-increase := 15000
+cp-increase := 5308
 core-tolerance := 0.02
 var-decay := 0.95
 cla-decay := 0.999
@@ -9,13 +9,13 @@
 phase-saving := 2
 rnd-init := 'False'
 luby := 0
-gc-frac := 0.2
+gc-frac := 0.21553442609855578
 verb := 1
 pre := 'True'
 asymm := 'False'
 rcheck := 'False'
 elim := 'False'
 grow := 0
-cl-lim := 20
+cl-lim := 0
 sub-lim := 1000
-simp-gc-frac := 0.5
+simp-gc-frac := 0.036323815755803346
