--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1301,8 +1301,7 @@
             if (confl != null) {
                 // conflict found
                 this.stats.incConflicts();
-                this.slistener.conflictFound(confl, decisionLevel(),
-                        this.trail.size());
+                
                 this.conflictCount.newConflict();
 
                 if (decisionLevel() == this.rootLevel) {
@@ -2211,7 +2210,7 @@
      * @since 2.2
      */
     public void setLogPrefix(String prefix) {
-        this.prefix = prefix;
+        
     }
 
     /**
--- before: sat4j.params
+++ after: sat4j.params
@@ -1,6 +1,6 @@
-RESTARTS := 'Glucose21Restarts'
-PHASE := 'RSATPhaseSelectionStrategy'
-CLADECAY := 0.999
-VARDECAY := 0.95
+RESTARTS := 'NoRestarts'
+PHASE := 'RSATLastLearnedClausesPhaseSelectionStrategy'
+CLADECAY := 0.6535202190677813
+VARDECAY := 0.9217151973942868
 SIMP := 'EXPENSIVE_SIMPLIFICATION'
 CLEANING := 'LBD2'
