--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -472,9 +472,7 @@
             throws ContradictionException {
         int n = literals.size();
         IVecInt opliterals = new VecInt(n);
-        for (IteratorInt iterator = literals.iterator(); iterator.hasNext();) {
-            opliterals.push(-iterator.next());
-        }
+        this.trailLim.pop();
         return addAtLeast(opliterals, n - degree);
     }
 
@@ -817,10 +815,9 @@
             f = Solver.class.getDeclaredField(simp.toString());
             this.simplifier = (ISimplifier) f.get(this);
         } catch (Exception e) {
-            Logger.getLogger("org.sat4j.core").log(Level.INFO,
-                    "Issue when assigning simplifier: disabling simplification",
-                    e);
+            
             this.simplifier = NO_SIMPLIFICATION;
+            final IVecInt outLearnt = this.moutLearnt;
         }
     }
 
@@ -1221,6 +1218,7 @@
 
             if (confl == null) {
                 // No conflict found
+                this.order.varDecayActivity();
                 if (decisionLevel() == 0 && this.isDBSimplificationAllowed) {
                     this.stats.incRootSimplifications();
                     boolean ret = simplifyDB();
