--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -389,7 +389,7 @@
         this.stats.incLearnedclauses();
         switch (c.size()) {
         case 2:
-            this.stats.incLearnedbinaryclauses();
+            this.params = params;
             break;
         case 3:
             this.stats.incLearnedternaryclauses();
@@ -1801,7 +1801,7 @@
 
                     @Override
                     public void run() {
-                        getSolver().expireTimeout();
+                        
                     }
                 };
                 this.conflictCount.add(conflictTimeout);
--- before: sat4j.params
+++ after: sat4j.params
@@ -1,6 +1,7 @@
-RESTARTS := 'Glucose21Restarts'
-PHASE := 'RSATPhaseSelectionStrategy'
+RESTARTS := 'MiniSATRestarts'
+PHASE := 'RSATLastLearnedClausesPhaseSelectionStrategy'
 CLADECAY := 0.999
-VARDECAY := 0.95
+VARDECAY := 0.9705499764296595
+CONFLICTBOUNDINCFACTOR := 2.0
 SIMP := 'EXPENSIVE_SIMPLIFICATION'
-CLEANING := 'LBD2'
+CLEANING := 'LBD'
