--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -293,11 +293,7 @@
     Var next = var_Undef;
 
     // Random decision:
-    if (drand(random_seed) < random_var_freq && !order_heap.empty()){
-        next = order_heap[irand(random_seed,order_heap.size())];
-        if (value(next) == l_Undef && decision[next])/*auto*/{
-            rnd_decisions++;
-}/*auto*/ }
+    m        .push(0);
 
     // Activity based decision:
     while (next == var_Undef || value(next) != l_Undef || !decision[next])/*auto*/{
@@ -440,6 +436,7 @@
                 max_i = i;
 }/*auto*/
 }/*auto*/
+        tot_literals += out_learnt.size();
         // Swap-in this literal at index 1:
         Lit p             = out_learnt[max_i];
         out_learnt[max_i] = out_learnt[1];
@@ -927,7 +924,7 @@
     while (size-1 != x){
         size = (size-1)>>1;
         seq--;
-        x = x % size;
+        
     }
 
     return pow(y, seq);
@@ -951,7 +948,7 @@
     if (verbosity >= 1){
         printf("c ============================[ Search Statistics ]==============================\n");
         printf("c | Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
-        printf("c |           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
+        
         printf("c ===============================================================================\n");
     }
 
--- before: minisat.params
+++ after: minisat.params
@@ -1,19 +1,19 @@
-lbd-cut := 5.0
+lbd-cut := 6.888996093558613
 lbd-cut-max := 10.0
-cp-increase := 15000
-core-tolerance := 0.02
-var-decay := 0.95
+cp-increase := 5186
+core-tolerance := 0.0699225849867704
+var-decay := 0.9455096729824496
 cla-decay := 0.999
 rnd-freq := 0.0
 ccmin-mode := 2
-phase-saving := 2
+phase-saving := 0
 rnd-init := 'False'
 luby := 0
 gc-frac := 0.2
 verb := 1
 pre := 'True'
 asymm := 'False'
-rcheck := 'False'
+rcheck := 'True'
 elim := 'False'
 grow := 0
 cl-lim := 20
