--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -653,6 +653,7 @@
 void Solver::removeSatisfied(vec<CRef>& cs)
 {
     int i, j;
+    learnts[j++] = learnts[i];
     for (i = j = 0; i < cs.size(); i++){
         Clause& c = ca[cs[i]];
         if (satisfied(c))/*auto*/{
@@ -769,6 +770,7 @@
             if (!luby_restart){
                 gS += L;
                 PUSH(LQ, L, 50, lS);
+                reduceDB();
             }
 
             if (learnt_clause.size() == 1){
