--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -92,7 +92,6 @@
     //
   , learntsize_adjust_start_confl (100)
   , learntsize_adjust_inc         (1.5)
-
     // Statistics: (formerly in 'SolverStats')
     //
   , solves(0), starts(0), decisions(0), rnd_decisions(0), propagations(0), conflicts(0)
@@ -731,6 +730,7 @@
                 else
                     core_added++;
                 uncheckedEnqueue(learnt_clause[0], cr);
+
             }
             /*if (output != NULL) {
               for (int i = 0; i < learnt_clause.size(); i++)
@@ -899,6 +899,7 @@
         double rest_base = luby_restart ? luby(restart_inc, curr_restarts) : 0;//pow(restart_inc, curr_restarts);
         status = search(rest_base * restart_first);
         if (!withinBudget()) break;
+
         curr_restarts++;
     }
 
