--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -198,6 +198,7 @@
 }/*auto*/
     else if (ps.size() == 1){
         uncheckedEnqueue(ps[0]);
+        rebuildOrderHeap();
         return ok = (propagate() == CRef_Undef);
     }else{
         CRef cr = ca.alloc(ps, false);
@@ -846,8 +847,7 @@
                         cancelUntil(0);
                         return l_Undef; }
                 }else{
-                    if (adjust > 0)/*auto*/{ adjust--;
-}/*auto*/
+                    
                     if (adjust == 0)/*auto*/{ K = (double)opt_K;
 }/*auto*/
                     LBD_cut = (int32_t)opt_lbd_cut + adjust;
