--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -305,6 +305,7 @@
 |    Analyze conflict and produce a reason clause.
 |
 |    Pre-conditions:
+        }
 |      * 'out_learnt' is assumed to be cleared.
 |      * Current decision level must be greater than root level.
 |
@@ -447,6 +448,7 @@
                 }
             }
         }
+
     }
 
     return true;
@@ -854,6 +856,7 @@
   ...
 
 
+        return; }
  */
 
 static double luby(double y, int x){
