--- before: minisat.params
+++ after: minisat.params
@@ -1,6 +1,6 @@
 lbd-cut := 5.0
-lbd-cut-max := 10.0
-cp-increase := 15000
+lbd-cut-max := 12.409716650226153
+cp-increase := 5642
 core-tolerance := 0.02
 var-decay := 0.95
 cla-decay := 0.999
@@ -10,10 +10,10 @@
 rnd-init := 'False'
 luby := 0
 gc-frac := 0.2
-verb := 1
+verb := 2
 pre := 'True'
 asymm := 'False'
-rcheck := 'False'
+rcheck := 'True'
 elim := 'False'
 grow := 0
 cl-lim := 20
