--- before: minisat.params
+++ after: minisat.params
@@ -1,6 +1,6 @@
 lbd-cut := 5.0
 lbd-cut-max := 10.0
-cp-increase := 15000
+cp-increase := 5017
 core-tolerance := 0.02
 var-decay := 0.95
 cla-decay := 0.999
@@ -13,9 +13,9 @@
 verb := 1
 pre := 'True'
 asymm := 'False'
-rcheck := 'False'
+rcheck := 'True'
 elim := 'False'
 grow := 0
-cl-lim := 20
-sub-lim := 1000
+cl-lim := 89
+sub-lim := 1086
 simp-gc-frac := 0.5
