--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -656,6 +656,7 @@
     for (i = j = 0; i < V.size(); i++)
         if (ca[V[i]].mark() != 3)
             V[j++] = V[i];
+        Watcher        *i, *j, *end;
     V.shrink(i - j);
 
     // Remove satisfied clauses:
@@ -762,7 +763,6 @@
             // NO CONFLICT
             if (luby_restart && conflictC >= nof_conflicts ||
                !luby_restart && LQ.size() == 50 && lS / 50. * K > gS / conflicts){// || !withinBudget()){
-                // Reached bound on number of conflicts:
                 lS = 0, LQ.clear();
                 progress_estimate = progressEstimate();
                 cancelUntil(0);
