--- before: minisat.params
+++ after: minisat.params
@@ -1,6 +1,6 @@
-lbd-cut := 5.0
-lbd-cut-max := 10.0
-cp-increase := 15000
+lbd-cut := 5.57854779560755
+lbd-cut-max := 21.535517722910527
+cp-increase := 5945
 core-tolerance := 0.02
 var-decay := 0.95
 cla-decay := 0.999
@@ -13,9 +13,9 @@
 verb := 1
 pre := 'True'
 asymm := 'False'
-rcheck := 'False'
-elim := 'False'
+rcheck := 'True'
+elim := 'True'
 grow := 0
 cl-lim := 20
 sub-lim := 1000
-simp-gc-frac := 0.5
+simp-gc-frac := 0.2550177094877917
