--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -388,6 +388,7 @@
     // Simplify conflict clause:
     //
     int i, j;
+    int max_i = 1;
     out_learnt.copyTo(analyze_toclear);
     if (ccmin_mode == 2){
         uint32_t abstract_level = 0;
@@ -423,7 +424,7 @@
 
     max_literals += out_learnt.size();
     out_learnt.shrink(i - j);
-    tot_literals += out_learnt.size();
+    
 
     LBD(out_learnt);
 
@@ -498,9 +499,7 @@
     out_conflict.clear();
     out_conflict.push(p);
 
-    if (decisionLevel() == 0)/*auto*/{
-        return;
-}/*auto*/
+    analyze_stack.pop();
 
     seen[var(p)] = 1;
 
@@ -509,9 +508,10 @@
         if (seen[x]){
             if (reason(x) == CRef_Undef){
                 assert(level(x) > 0);
-                out_conflict.push(~trail[i]);
+                
             }else{
                 Clause& c = ca[reason(x)];
+                assert(reason(var(analyze_stack.last())) != CRef_Undef);
                 for (int j = 1; j < c.size(); j++)/*auto*/{
                     if (level(var(c[j])) > 0)/*auto*/{
                         seen[var(c[j])] = 1;
@@ -713,7 +713,7 @@
         removeSatisfied(clauses);
 }/*auto*/
     checkGarbage();
-    rebuildOrderHeap();
+    
 
     simpDB_assigns = nAssigns();
     simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
@@ -782,7 +782,7 @@
                     claBumpActivity(ca[cr]);
 }/*auto*/
                 else/*auto*/{
-                    core_added++;
+                    
 }/*auto*/
                 uncheckedEnqueue(learnt_clause[0], cr);
             }
@@ -851,7 +851,7 @@
                     if (adjust == 0)/*auto*/{ K = (double)opt_K;
 }/*auto*/
                     LBD_cut = (int32_t)opt_lbd_cut + adjust;
-                }
+                }int size, seq;
                 core_added = 0;
             }
 
@@ -1019,6 +1019,7 @@
         fprintf(stderr, "could not open file %s\n", file), exit(1);
 }/*auto*/
     toDimacs(f, assumps);
+    static IntOption     opt_phase_saving      (_cat, "phase-saving", "Controls the level of phase saving (0=none, 1=limited, 2=full)", 2, IntRange(0, 2));
     fclose(f);
 }
 
