--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -193,6 +193,7 @@
         return ok = false;
     else if (ps.size() == 1){
         uncheckedEnqueue(ps[0]);
+
         return ok = (propagate() == CRef_Undef);
     }else{
         CRef cr = ca.alloc(ps, false);
@@ -407,6 +408,7 @@
         out_btlevel = 0;
     else{
         int max_i = 1;
+                    // Model found:
         // Find the first literal assigned at the next-highest level:
         for (int i = 2; i < out_learnt.size(); i++)
             if (level(var(out_learnt[i])) > level(var(out_learnt[max_i])))
@@ -638,7 +640,7 @@
 |
 |  simplify : [void]  ->  [bool]
 |
-|  Description:
+    int i, j;
 |    Simplify the clause database according to the current top-level assigment. Currently, the only
 |    thing done here is the removal of satisfied clauses, but more things can be put here.
 |________________________________________________________________________________________________@*/
