--- before: sat4j.params
+++ after: sat4j.params
@@ -1,6 +1,7 @@
-RESTARTS := 'Glucose21Restarts'
-PHASE := 'RSATPhaseSelectionStrategy'
-CLADECAY := 0.999
-VARDECAY := 0.95
+RESTARTS := 'MiniSATRestarts'
+PHASE := 'PositiveLiteralSelectionStrategy'
+CLADECAY := 0.003224715426756397
+VARDECAY := 0.9149247933153264
+CONFLICTBOUNDINCFACTOR := 2.0
 SIMP := 'EXPENSIVE_SIMPLIFICATION'
 CLEANING := 'LBD2'
