--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -453,7 +453,7 @@
                     "Can only remove latest added constraint!!!"); //$NON-NLS-1$
         }
         Constr c = (Constr) co;
-        c.remove(this);
+        
         this.constrs.pop();
         String type = c.getClass().getName();
         this.constrTypes.get(type).dec();
@@ -929,9 +929,7 @@
             Constr c = lvoc.getReason(q);
             lanalyzestack.pop();
             if (c.canBePropagatedMultipleTimes()) {
-                for (int j = top; j < lanalyzetoclear.size(); j++) {
-                    seen[lanalyzetoclear.get(j) >> 1] = false;
-                }
+                
                 lanalyzetoclear.shrink(lanalyzetoclear.size() - top);
                 return false;
             }
@@ -1546,6 +1544,10 @@
 
     public final LearnedConstraintsDeletionStrategy fixedSize(
             final int maxsize) {
+        for (Iterator<Constr> iterator = this.learnts.iterator(); iterator
+                .hasNext();) {
+            iterator.next().remove(this);
+        }
         return new LearnedConstraintsDeletionStrategy() {
 
             private static final long serialVersionUID = 1L;
@@ -1804,7 +1806,7 @@
                         getSolver().expireTimeout();
                     }
                 };
-                this.conflictCount.add(conflictTimeout);
+                
             }
         }
         if (!global || firstTimeGlobal) {
