--- before: sat4j.params
+++ after: sat4j.params
@@ -1,6 +1,7 @@
-RESTARTS := 'Glucose21Restarts'
-PHASE := 'RSATPhaseSelectionStrategy'
-CLADECAY := 0.999
+RESTARTS := 'MiniSATRestarts'
+PHASE := 'RSATLastLearnedClausesPhaseSelectionStrategy'
+CLADECAY := 0.7097918006890688
 VARDECAY := 0.95
-SIMP := 'EXPENSIVE_SIMPLIFICATION'
+CONFLICTBOUNDINCFACTOR := 2.0
+SIMP := 'SIMPLE_SIMPLIFICATION'
 CLEANING := 'LBD2'
