--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -598,10 +598,7 @@
         "output-out-of-bag-complexity-statistics", 0,
         "-output-out-of-bag-complexity-statistics"));
 
-    newVector
-      .addElement(new Option(
-        "\tPrint the individual classifiers in the output", "print", 0,
-        "-print"));
+    
 
     newVector.addElement(new Option(
       "\tCompute and output attribute importance (mean impurity decrease "
--- before: weka.params
+++ after: weka.params
@@ -5,6 +5,6 @@
 V := 0.001
 depth := 0
 N := 0
-U := 'False'
+U := 'True'
 B := 'False'
 num-decimal-places := 2.0
